src/profiler.cpp

```diff
@@ -1075,6 +1088,30 @@ Error Profiler::start(Arguments& args, bool reset) {
     return error;
 }
 
+Error Profiler::setEcid(const char* ecid) {
+    pthread_once(&local_ecid_key_once, createEcidKey);
+
+    void *value = pthread_getspecific(local_ecid_key);
+    destroyEcidKey(value);
+
+    int size = strlen(ecid);
+    char *duplicate = (char*) malloc( size+ 1 );
+    strcpy(duplicate, ecid);
```

Comment:
You already know the size, so you can use `memcpy` to avoid overhead of calculating the string size again (which is done inside the `strcpy`).

Reply:
`strdup`

Reply:
or just [`strdup`](https://man7.org/linux/man-pages/man3/strdup.3.html)

Reply:
Code is not longer in use, resolving conversation.